src/WikiParsing/WikiString.fs

```diff
@@ -2,44 +2,91 @@
 
 open StringHelper
 
-let formSeparators = [|'/'; ','|]
+type private Form = {
+    Label: string option
+    Word: string
+}
 
-let allowedLabels  = [
-    "(řidč.)"
-    "(zřídka)"
-    "(knižně)"
+let private formSeparators = [|'/'; ','|]
+let private referencePattern = "\[(p )?\d*\]"
+let private allowedLabels  = [
+     "řidč."
+     "zřídka"
+     "knižně"
     ]
 
-let rejectedLabels = [
-    "(zastarale)"
-    "(hovorově)"
-    "(v obecném jazyce)"
-    "(archaicky)"
-    "(básnicky)"
-    "(nářečně)"
-    "*"
-    ]
+let private isBlank s = s = "" || s = "—" || s = "-"
+let private meansUnused = ends "*"
+let private isPlainWord = isMatch "^\w+[\s\-]?\w+$"
+let private isSpaced s = s |> starts " " || s |> ends " " 
+let private hasReferences = contains "["
+let private hasLabel = starts "("
+let private hasSecondFormAfterColon = contains ":"
+let private hasSplitForms = containsAny (formSeparators |> Seq.map string)
+
+let private getLabeledForm =
+    ``match`` "^\((?<label>.*)\) (?<word>\w*)$"
+    >> fun m ->
+    { 
+        Word = m.Groups.["word"].Value
+        Label = Some m.Groups.["label"].Value
+    }
+
+let private getFormsWhenColon =
+    ``match`` "^(?<word1>\w*) \((?<label>\w*): (?<word2>\w*)\)$"
+    >> fun m -> [|
```

Comment:
Why do we need arrays here / not lists?

Reply:
We don't need arrays anymore. I even tried to replace them but that led to some changes in the code using this stuff so I wanted to send that a separate change.